src/renderer/actions/repair.js

```diff
@@ -14,76 +13,128 @@ import kill from "./utils/kill";
-    for (const channel in config) {
-        const roaming = path.join(remote.app.getPath("userData"), "..", platforms[channel].replace(" ", "").toLowerCase());
+    for (const nowProcessing of toProcess) {
+        // Calculate the platform-specific Discord config path.
+        const discordConfigPath = (isLinux)
+            ? path.join(nowProcessing, "../../..")  // Jump out to the Discord installation's main xdg-config sub-directory.
+            : path.join(remote.app.getPath("userData"), "..", platforms[nowProcessing].replace(" ", "").toLowerCase());
+
         try {
-            const versionDir = (await fs.readdir(roaming)).find(d => d.split(".").length > 2);
-            const modulesPath = path.join(roaming, versionDir, "modules");
-            log("Removing " + modulesPath);
+            // Detect the latest Discord version directory in this path.
+            const version = findLatestDiscordVersion(discordConfigPath);
+            if (!version) continue;  // Skip path since nothing was found.
+
+            // Cross-platform: Remove the "modules" directory for that version.
+            // NOTE: This will cause Discord to redownload the latest version again on next launch.
+            const modulesPath = path.join(discordConfigPath, version, "modules");
             if (await exists(modulesPath)) {
+                log(`Removing ${modulesPath}.`);
                 const error = await new Promise(resolve => rimraf(path.join(modulesPath), originalFs, resolve));
                 if (error) {
-                    log(`❌ Could not delete modules in ${roaming}`);
+                    log(`❌ Could not delete modules in ${discordConfigPath}.`);
                     log(`❌ ${error.message}`);
                     return error;
                 }
+                log("✅ Deletion successful.");
             }
-            log("✅ Deletion successful");
             progress.set(progress.value + progressPerLoop);
         }
         catch (err) {
-            log(`❌ Could not delete modules in ${roaming}`);
+            log(`❌ Could not delete modules in ${discordConfigPath}.`);
             log(`❌ ${err.message}`);
             return err;
         }
     }
 }
 
-async function showInstallNotice(config) {
-    const confirmation = await remote.dialog.showMessageBox(remote.BrowserWindow.getFocusedWindow(), {
-        type: "question",
-        title: "Reinstall BetterDiscord?",
-        message: "After repairing, you need to reinstall BetterDiscord. Would you like to do that now?",
-        noLink: true,
-        cancelId: 1,
-        buttons: ["Yes", "No"]
-    });
 
-    if (confirmation.response !== 0) return succeed();
+async function showInstallNotice(config, discordPaths) {
+    // Let the user know that they must launch Discord to download the "modules"
+    // folders that we deleted, otherwise we won't be able to inject since the
+    // files won't exist. If the user proceeds without reading, they'll get an error.
+    while (true) {
```

Comment:
Better to use an awaited promise than a `while(true)`

Reply:
Could you explain an example of that? I know what promises are, but what kind of promise would I be awaiting here? You might be thinking of some Javascript pattern that I am not aware of. Some kind of "infinite await/loop promise"?

Reply:
Alright I looked into it. The code pattern I use is correct already.

- The function itself is marked async.
- `while(true)` is totally normal in async functions, as long as you do some `await` actions inside the loop to pass away control.
- We constantly pass away control via `await` inside the `while(true)` loop, for things like the filesystem checks and waiting for the dialog box responses.

So it's already correct. See:

https://stackoverflow.com/questions/39110762/while-loops-using-await-async